pkg/deployer/manifest/objectapplier.go

```diff
@@ -142,43 +146,36 @@ func (a *ObjectApplier) ApplyObject(ctx context.Context, i int, manifestData man
 }
 
 // cleanupOrphanedResources removes all managed resources that are not rendered anymore.
-func (a *ObjectApplier) cleanupOrphanedResources(ctx context.Context) error {
+func (a *ObjectApplier) cleanupOrphanedResources(ctx context.Context, mr []manifest.ManagedResourceStatus) error {
 	var (
 		allErrs []error
 		wg      sync.WaitGroup
 	)
 
-	for _, mr := range a.managedResources {
+	for _, mr := range mr {
```

Comment:
we should rename mr to avoid confusion.
```suggestion
	for _, mr := range oldManagedResources {
```

Reply:
Done.